independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/contexts/request/propagation/RequestContextPropagationTest.java

```diff
@@ -56,9 +58,21 @@ public void testPropagation() {
         } catch (ContextNotActiveException expected) {
         }
 
-        requestContext.activate(state);
+        requestContext.activate(state1);
         assertEquals(arc.instance(SuperController.class).get().getId(), controller2Id);
 
+        // add req. scoped bean, note that we have already captured the state
+        // this proves that the map is shared during context propagation
+        YetAnotherReqScopedBean yetAnotherReqScopedBean = arc.instance(YetAnotherReqScopedBean.class).get();
+        int generatedNumber = yetAnotherReqScopedBean.getRandomNumber();
+
+        // end the context and re-start with state1 and assume the bean is part of it
+        requestContext.deactivate();
+        requestContext.activate(state1);
+
+        // check that the bean is not created anew - that means the coxtextual storage was shared
+        assertEquals(arc.instance(YetAnotherReqScopedBean.class).get().getRandomNumber(), generatedNumber);
```

Comment:
This assert does not verify the `YetAnotherReqScopedBean` was propagated "back" to the `state1`. I think you should check the `state1.getContextualInstances()` for presence of the `YetAnotherReqScopedBean` key instead.

Reply:
It does verify it. If it weren't stored, new bean would be created with different random number.
Or I don't follow what you mean?
I did try this test firstly with the previous state of propagation and it failed here...

Reply:
I'll add some more comments to make it clearer.
It basically simulates the scenario where you take a context snapshot (capture a state) and after that you create another bean - the test verifies that even such a bean gets to be propagated.